tls/extensions/s2n_client_supported_groups.c

```diff
@@ -19,18 +19,23 @@
 #include "tls/extensions/s2n_client_supported_groups.h"
 #include "tls/s2n_tls.h"
 #include "tls/s2n_tls_parameters.h"
+#include "tls/s2n_ecc_preferences.h"
 
 #include "utils/s2n_safety.h"
 
 int s2n_extensions_client_supported_groups_send(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
+    notnull_check(conn);
+    const struct s2n_ecc_preferences *ecc_pref = conn->config->ecc_preferences;
+    notnull_check(ecc_pref);
+
     GUARD(s2n_stuffer_write_uint16(out, TLS_EXTENSION_SUPPORTED_GROUPS));
-    GUARD(s2n_stuffer_write_uint16(out, 2 + s2n_ecc_evp_supported_curves_list_len * 2));
+    GUARD(s2n_stuffer_write_uint16(out, 2 + ecc_pref->count * 2));
```

Comment:
Can you add a comment for the hardcoded values (size of extension, 2 byte iana ids)